diesel/src/pg/expression/mod.rs

```diff
@@ -1,5 +1,7 @@
 #[doc(hidden)]
 pub mod array_comparison;
+#[doc(hidden)]
+pub mod array;
```

Comment:
Does this need to be public?

Reply:
Now that `IntoSingleTypeExpressionList` is in that module, yes.

Reply:
Can we have a `#[doc(hidden)]` reexport of this from somewhere public and make the module private?

Reply:
I'm not exactly sure what shape this should take, having tried a few ways. I've tried something in 99783c00, but it feels janky and wrong. Some documentation added too.